### PR TITLE
Rand 0.5

### PR DESCRIPTION
**Pull request by antiguru**
Update to support rand 0.5. Also some cleanup.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -99,20 +97,20 @@ impl ZipfDistribution {
         // looks slightly different.
 
         // We know these were computed in new()
-        let hnum = self.h_integral_num_elements.unwrap();
-        let h_x1 = self.h_integral_x1.unwrap();
-        let s = self.s.unwrap();
+        let hnum = self.h_integral_num_elements;
+        let h_x1 = self.h_integral_x1;
+        let s = self.s;
```
Comment (jonhoo) [inline src/lib.rs]: Now that these are no longer `Option`, we can just use them inline below instead :)

File: src/lib.rs [hunk 2]
```diff
@@ -172,6 +170,12 @@ impl rand::distributions::Sample<usize> for ZipfDistribution {
     }
 }
 
+impl rand::distributions::Distribution<usize> for ZipfDistribution {
```
Comment (jonhoo) [inline src/lib.rs]: I wonder if we should also implement `IndependentSample` for completeness (even though it is deprecated).

**Comment by jonhoo** [comment]
These changes look great! Thank you. I've left two relatively minor inline comments — once those are resolved, let's merge and release :)

**Comment by antiguru** [comment]
I addressed your comments and did some more cleanup.

**Comment by jonhoo** [comment]
Great, thank you!

**Comment by jonhoo** [comment]
Published in `3.1.0`!

**Pull request merged**